cmd/dogstatsd/subcommands/start/command.go

```diff
@@ -240,7 +240,7 @@ func RunDogstatsd(_ context.Context, cliParams *CLIParams, config config.Compone
 		log.Warnf("Can't setup core dumps: %v, core dumps might not be available after a crash", err)
 	}
 
-	if !config.IsSet("api_key") {
+	if !config.IsKnown("api_key") {
```

Comment:
This looks wrong.

Based on my understanding of what a "known" key is, this will always be `true` since `api_key` is defined with an environment variable bound to it... and that's not what we're trying to assert here.

Reply:
from documentation and from code reading, It will check if the entry exists. so if this is not set, then it should return false, let my try it and I'll come back to you

Reply:
you are right, i used the wrong method, I changed it to a method that is not depreccated and has the expected scenario.